var/spack/repos/builtin/packages/papi/package.py

```diff
@@ -94,12 +94,13 @@ def setup_build_environment(self, env):
             env.set("HSA_TOOLS_LIB", "%s/librocprofiler64.so" % spec["rocprofiler-dev"].prefix.lib)
             env.append_flags("CFLAGS", "-I%s/rocprofiler/include" % spec["rocprofiler-dev"].prefix)
             env.set(
-                "ROCP_METRICS", "%s/rocprofiler/lib/metrics.xml" % spec["rocprofiler-dev"].prefix
+                "ROCP_METRICS", "%s/rocprofiler/metrics.xml" % spec["rocprofiler-dev"].prefix.lib
```

Comment:
```suggestion
                "ROCP_METRICS", "%s/metrics.xml" % spec["rocprofiler-dev"].prefix.lib
```

Reply:
This won't work because for rocm < 5.2.0 the location is `<path_to_rocprofiler>/rocprofiler/lib`, while for rocm >= 5.2.0 the location is `<path_to_rocprofiler>/lib/rocprofiler`. I have added an `os.path.exists()` to use the right format accordingly.